src/modules/fundingManager/depositVault/FM_DepositVault_v1.sol

```diff
@@ -94,6 +97,17 @@ contract FM_DepositVault_v1 is
         address from = _msgSender();
         token().safeTransferFrom(from, address(this), amount);
 
+        (uint collateralFeePercentage, address collateralTreasury) =
+        _getFeeManagerCollateralFeeData(
+            bytes4(keccak256(bytes("deposit(uint)")))
+        );
+
+        uint protocolFeeAmount = amount * collateralFeePercentage / BPS;
```

Comment:
It probably makes sense to have a similar flow here as within the [BondingCurveBase_v1](https://github.com/InverterNetwork/contracts/blob/main/src/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.sol#L357), i.e. check if calculation is needed and check for rounding down to zero situation.

Reply:
The _processProtocolFeeViaTransfer function could then also be moved into the if statement, removing the need for the check in the funciton.